### DISCUSSION TITLE
Leaf phenology parameters

### DISCUSSION THREAD

**Discussion by mpaiao** [NGEET/fates#956] [General]
In the parameter files, we have three PFT-dependent parameters to describe which type of leaf phenology strategy is associated with each PFT: `fates_phen_evergreen`, `fates_phen_season_decid`, `fates_phen_stress_decid`. The code checks and prevents a PFT to have more than one of these types simultaneously active.

I am actually adding a drought semi-deciduous leaf phenology to FATES, and I am wondering if we should simplify this by creating a single PFT-dependent parameter that describes the leaf phenology (`fates_phen_type` or something similar), with options like:
- 0: evergreen
- 1: cold deciduous
- 2: drought "hard" deciduous
- 3: drought semi-deciduous

This would allow us to add more strategies without multiplying the number of parameters and internal tests. I am not sure if there would be any downsides to this, though.

**Comment by glemieux** [comment]
This looks like a pretty good idea to me.  Would the idea be to add some simple code to handle assigning the associated `prt_param` with the corresponding true/false value?  If so, this would allow us to avoid refactoring the checks against these parameters.  

From a historical perspective, my assumption is that current convention was adopted by fates based on/inspired by the host land models vegetation properties type.  Is that correct?

**Comment by mpaiao** [comment]
I was actually thinking about refactoring the checks in the code and having a single `prt_param%phenology` variable instead (and having some public parameters `ievergreen`, `iseason_decid`, `ihard_stress_decid`, `isemi_stress_decid`, etc. for the checks). But I can see that this could make it harder to track changes, and I am happy to go either way.

Sorry, I don't know the source of the current convention...

**Reply by ckoven** [reply]
@mpaiao this all makes sense to me.

**Comment by rgknox** [comment]
This is great @mpaiao . I like the idea of defining ivergreen, etc.  EDTypesMod or FatesConstantsMod would both work to hold the definitions.  Both modules have similar types of definitions.

**Comment by rosiealice** [comment]
Great idea @mpaiao. I'm excited about semi deciduosness :) 

On the historical front, @glemieux, FATES didn't use to have any drought deciduousness, so yes, it was generally added to mirror the HLM convention (which was actually JULES at that point!)

**Reply by glemieux** [reply]
Thanks @rosiealice; I thought something must be up when I looked at the HLM code and saw the same naming convention in the veg types.

**Comment by mpaiao** [comment]
Thanks all for the feedback! Perhaps to keep things a bit more tractable, I may submit two pull requests, one with the changes in drought-deciduous phenology (#801, though I may need to supersede that with another PR) and one with the parameter change.
